lumigator/backend/backend/tracking/mlflow.py

```diff
@@ -59,53 +62,58 @@ async def create_experiment(
+        :param description: A description of the experiment.
+        :param task_definition: The ``TaskDefinition`` for the experiment (e.g. summarization, translation).
+        :param dataset: The dataset ID associated with the experiment.
+        :param max_samples: The maximum number of samples to use for the experiment.
+        :return: The experiment response containing the new experiment ID and other details.
+        :raises ExperimentConflictError: If the experiment name already exists.
+        :raises TrackingClientUpstreamError: If there is a problem with the tracking client.
+        """
+        experiment_name = name
+        tags = {
+            "description": description,
+            "task_definition": task_definition.model_dump_json(),
+            "dataset": str(dataset),
+            "max_samples": str(max_samples),
+            "lumigator_version": "0.2.1",
+        }
+
+        # Try to create the experiment and handle any conflicts
         try:
-            experiment_id = self._client.create_experiment(name)
-            self._client.set_experiment_tag(experiment_id, "description", description)
-            self._client.set_experiment_tag(experiment_id, "task_definition", task_definition.model_dump_json())
-            self._client.set_experiment_tag(experiment_id, "dataset", dataset)
-            self._client.set_experiment_tag(experiment_id, "max_samples", str(max_samples))
-            self._client.set_experiment_tag(experiment_id, "lumigator_version", "0.2.1")
-        # FIXME Let the user decide in the case of failures.
-        # Also, use a uuid as name and an arbitrary tag as descriptive name
-        except MlflowException as e:
-            # if the experiment name already exists,
-            # log a warning and then append a short timestamp to the name
-            if "RESOURCE_ALREADY_EXISTS" in str(e):
-                print(f"Experiment name '{name}' already exists. Appending timestamp.")
-                name = f"{name} {datetime.now().strftime('%Y%m%d%H%M%S')}"
-                experiment_id = self._client.create_experiment(name)
-                self._client.set_experiment_tag(experiment_id, "description", description)
-                self._client.set_experiment_tag(experiment_id, "task_definition", task_definition.model_dump_json())
-                self._client.set_experiment_tag(experiment_id, "dataset", dataset)
-                self._client.set_experiment_tag(experiment_id, "max_samples", str(max_samples))
-                self._client.set_experiment_tag(experiment_id, "lumigator_version", "0.2.1")
-            else:
-                raise e
+            experiment_id = self._create_experiment(experiment_name, tags)
+        except ExperimentConflictError:
+            # FIXME Let the user decide in the case of failures.
+            # Also, use a uuid as name and an arbitrary tag as descriptive name
+            timestamp_suffix = datetime.now().strftime("%Y%m%d%H%M%S%f")
```

Comment:
```suggestion
            timestamp_suffix = datetime.now().strftime("%Y%m%d%H%M%S")
```
To have the same timestamp format as before?

Reply:
This change adds more precision to the timestamp as I had some tests fail on me because they created experiments in the same second :'(